pkg/steps/template.go

```diff
@@ -434,6 +434,14 @@ func waitForPodDeletion(podClient coreclientset.PodInterface, name string, uid t
 		if pod.UID != uid {
 			return nil
 		}
+		for j := 0; j < timeout; j += 2 {
```

Comment:
This does not really look right to me... This cycle will call `podRunningContainers(pod)` 150 times in immediate succession on each iteration of the outer cycle. which seems excessive and not useful. Also, isn't `pod` invariant here?

Reply:
>This cycle will call podRunningContainers(pod) 150 times in immediate succession on each iteration of the outer cycle

Oops, sorry, added `time.Sleep(2 * time.Second)` to wait for teardown container to complete.

This still doesn't look optimal to me though. I wonder if we can setup a watch for pods instead of fetching pod details every time?

Reply:
Yeah, two nested wait loops do not seem that optimal to me either - worst case, this could wait what, 2*150*300 seconds? That's more than a day... 

But still, does that cycle even make sense? To me this looks like `podRunningContainers(pod)` would return the same result on each iteration, `podRunningContainers` seems to work with just a Go object - it won't change between the nested cycle iterations. 

This will need something smarter, imho, just like you suggest.

Reply:
Ah, right, `podRunningContainers` includes Terminated